platform/admin/configurator/admin-configurator-actions-api/src/main/java/org/codice/ddf/internal/admin/configurator/actions/ConfigActions.java

```diff
@@ -13,10 +13,59 @@
  **/
 package org.codice.ddf.internal.admin.configurator.actions;
 
+import java.nio.file.Path;
+import java.util.Dictionary;
+
+import org.codice.ddf.admin.configurator.ConfiguratorException;
+import org.codice.ddf.admin.configurator.Operation;
+
 /**
- * Property operations adapted for
+ * Config operations adapted for the Configurator API.
+ * <p>
  * <b> This code is experimental. While this interface is functional and tested, it may change or be
  * removed in a future version of the library. </b>
  */
-public interface ConfigActions extends PropertyActions {
+public interface ConfigActions {
+    /**
+     * Creates a handler for persisting config file changes to a new config file.
+     *
+     * @param configFile the config file to be created
+     * @param configs    dictionary of key:value pairs to be written to the config file
+     * @return instance of this class
+     * @throws ConfiguratorException if an error occurs creating the operator
+     */
+    Operation<Void> create(Path configFile, Dictionary<String, Object> configs)
+            throws ConfiguratorException;
+
+    /**
+     * Creates a handler for deleting a config file.
+     *
+     * @param configFile the config file to be deleted
+     * @return instance of this class
+     * @throws ConfiguratorException if an error occurs creating the operator
+     */
+    Operation<Void> delete(Path configFile) throws ConfiguratorException;
+
+    /**
+     * Creates a handler for persisting config file changes to an existing config file.
+     *
+     * @param configFile       the config file to be updated
+     * @param configs          dictionary of key:value pairs to be written to the config file
+     * @param keepIfNotPresent if true, any keys in the current config file that are not in the
+     *                         {@code configs} map will be left with their initial values; if false, they
+     *                         will be removed from the file
+     * @return instance of this class
+     * @throws ConfiguratorException if an error occurs creating the operator
+     */
+    Operation<Void> update(Path configFile, Dictionary<String, Object> configs,
+            boolean keepIfNotPresent) throws ConfiguratorException;
```

Comment:
✏️ Should we even keep this `keepIfNotPresent` property around if we're only using it for the case that it's false? We could just document that the behavior is all additional properties will be removed.